src/libre/re.c

```diff
@@ -40,7 +40,7 @@ re_dialect(enum re_dialect dialect)
 		{ RE_LITERAL, parse_re_literal, 0, RE_SINGLE | RE_ANCHORED },
 		{ RE_GLOB,    parse_re_glob,    0, RE_SINGLE | RE_ANCHORED },
 		{ RE_NATIVE,  parse_re_native,  0, 0                       },
-		{ RE_PCRE,    parse_re_pcre,    0, 0                       },
+		{ RE_PCRE,    parse_re_pcre,    0, RE_END_NL               },
```

Comment:
Should we have `RE_END_NL` for the native dialect too? I'm not sure.

Reply:
I didn't want to assume. Personally I find the PCRE newline/$ behavior to be really awkward. It may be a special case due to interface reasons, but it introduces a bunch of other special cases with surprising consequences.

If you want to try, setting the flag there and making changes in the native parser similar to PCRE's (https://github.com/katef/libfsm/blob/sv/updates-to-pcre-tests/src/libre/dialect/pcre/parser.sid#L648-L650) should be sufficient, I think.